aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -193,7 +195,15 @@ module aptos_framework::transaction_validation {
             coin::balance<AptosCoin>(addr) >= transaction_fee_amount,
             error::out_of_range(PROLOGUE_ECANT_PAY_GAS_DEPOSIT),
         );
-        transaction_fee::burn_fee(addr, transaction_fee_amount);
+
+        if (features::collect_and_distribute_gas_fees()) {
+            // If transaction fees are redistributed to validators, collect them here for
+            // later redistribution.
+            transaction_fee::collect_fee(addr, transaction_fee_amount);
+        } else {
```

Comment:
Can you leave a TODO here to delete this else branch once transaction_fee has been launched and is fully proven to work well?